ui/src/components/canvas/FlowCanvas.tsx

```diff
@@ -120,11 +135,58 @@ const FlowCanvas = () => {
           onPaneClick={() => clearSelectedChildNode()}
           fitView
         >
-          <Controls>
+          <Controls style={{ bottom: "50px" }}>
+            <ControlButton
+              title="horizontal layout"
+              onClick={() =>
+                updateLayoutOrientation(
+                  "horizontal",
+                  getGraphConstraintOrientation()
```

Comment:
The layout density setting already exists in the store, so you don't need to pass it into the update method.